### DISCUSSION TITLE
CSS snippets

### DISCUSSION THREAD

**Discussion by charle36** [mdelobelle/metadatamenu#678] [Show and tell]
Hey everyone! 

I'm a CSS noob, but I played with it a little bit and corrected some stuff I didn't like in how some elements are displayed.

Here's the code: 
```
/* Correct the Add Type Button's position */
.fileClass-add-button {
	margin-top: auto;
	margin-bottom: auto;
}

/* Correct the position of the icon in the header in stacking mode */
.workspace-tab-header-inner .metadata-menu.fileclass-icon {
	margin-right: auto;
	margin-left: auto;
	order: 2;
}


/* Make the icon a prefix in dataview (from the forum) */
.dataview.table-view-table > tbody > tr > td:first-child > span{
  display: grid;
  grid-template-columns: auto auto;
  float: left;
}

.dataview.table-view-table > tbody > tr > td:first-child > span > a.internal-link{
  grid-column: 2;
}

.dataview.table-view-table > tbody > tr > td:first-child > span > a.fileclass-icon{
  grid-area: 1;
  margin-right: 0.375em;
}

/* Make the icon a prefix in the file explorer */
.tree-item-self.is-clickable.nav-file-title {
  display: grid;
  grid-template-columns: auto auto auto auto;
  float: left;
}

.tree-item-self.is-clickable.nav-file-title.content {
 	grid-column: 2;
}

.tree-item.nav-file .metadata-menu.fileclass-icon {
	margin-top: auto;
	margin-bottom: auto;
 	grid-column: 1;
	grid-row: 1;
	margin-right: 0.375em;
}
```

I can't find how to make the icon a prefix in internal links. If you have any idea how to do it, it would be welcome!

**Comment by LynetteCullens** [comment]
> Hey everyone!
> 
> I'm a CSS noob, but I played with it a little bit and corrected some stuff I didn't like in how some elements are displayed.
> 
> Here's the code:
> 
> ```
> /* Correct the Add Type Button's position */
> .fileClass-add-button {
> 	margin-top: auto;
> 	margin-bottom: auto;
> }
> 
> /* Correct the position of the icon in the header in stacking mode */
> .workspace-tab-header-inner .metadata-menu.fileclass-icon {
> 	margin-right: auto;
> 	margin-left: auto;
> 	order: 2;
> }
> 
> 
> /* Make the icon a prefix in dataview (from the forum) */
> .dataview.table-view-table > tbody > tr > td:first-child > span{
>   display: grid;
>   grid-template-columns: auto auto;
>   float: left;
> }
> 
> .dataview.table-view-table > tbody > tr > td:first-child > span > a.internal-link{
>   grid-column: 2;
> }
> 
> .dataview.table-view-table > tbody > tr > td:first-child > span > a.fileclass-icon{
>   grid-area: 1;
>   margin-right: 0.375em;
> }
> 
> /* Make the icon a prefix in the file explorer */
> .tree-item-self.is-clickable.nav-file-title {
>   display: grid;
>   grid-template-columns: auto auto auto auto;
>   float: left;
> }
> 
> .tree-item-self.is-clickable.nav-file-title.content {
>  	grid-column: 2;
> }
> 
> .tree-item.nav-file .metadata-menu.fileclass-icon {
> 	margin-top: auto;
> 	margin-bottom: auto;
>  	grid-column: 1;
> 	grid-row: 1;
> 	margin-right: 0.375em;
> }
> ```
> 
> I can't find how to make the icon a prefix in internal links. If you have any idea how to do it, it would be welcome!

Weird Folder Stacking Occurs 

![image](https://github.com/mdelobelle/metadatamenu/assets/45722611/deadd9e5-589b-49df-a1b1-24998ceceb57)
Updated snippet to prevent it from happening
```
/* Correct the Add Type Button's position */
.fileClass-add-button {
	margin-top: auto;
	margin-bottom: auto;
}

/* Correct the position of the icon in the header in stacking mode */
.workspace-tab-header-inner .metadata-menu.fileclass-icon {
	margin-right: auto;
	margin-left: auto;
	order: 2;
}


/* Make the icon a prefix in dataview (from the forum) */
.dataview.table-view-table > tbody > tr > td:first-child > span{
  display: grid;
  grid-template-columns: auto auto;
}

.dataview.table-view-table > tbody > tr > td:first-child > span > a.internal-link{
  grid-column: 2;
}

.dataview.table-view-table > tbody > tr > td:first-child > span > a.fileclass-icon{
  grid-area: 1;
  margin-right: 0.375em;
}

/* Make the icon a prefix in the file explorer */
.tree-item-self.is-clickable.nav-file-title {
  display: grid;
  grid-template-columns: auto auto auto auto;
}

.tree-item-self.is-clickable.nav-file-title.content {
 	grid-column: 2;
}

.tree-item.nav-file .metadata-menu.fileclass-icon {
	margin-top: auto;
	margin-bottom: auto;
 	grid-column: 1;
	grid-row: 1;
	margin-right: 0.375em;
}

/* Remove float: left to prevent pushing */
.dataview.table-view-table > tbody > tr > td:first-child > span {
  float: none; /* Remove float */
}

.tree-item-self.is-clickable.nav-file-title {
  float: none; /* Remove float */
}
```
